File: eva-core/eva-core-common/src/main/java/io/nerv/core/enums/LockEnumm.java
package io.nerv.core.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author: S.PKAQ
 */
@Getter
@AllArgsConstructor
public enum LockEnumm implements BizCode{
    /**
     * 锁定
     */
    LOCK("锁定", "0001"),
    UNLOCK("正常", "0000");

    /**
     * 名称
     */
    private String name;
    /**
     * 索引
     */
    private String index;
}

File: eva-core/src/main/kotlin/io/nerv/core/enums/LockEnumm.kt
package io.nerv.core.enums

/**
 * @author: S.PKAQ
 */
enum class LockEnumm(var value: String, var code: String) : BizCode {
    /**
     * 锁定
     */
    LOCK("锁定", "0001"),
    UNLOCK("正常", "0000");

    override fun getName(): String? {
        return this.value
    }

    override fun getIndex(): String? {
        return this.code
    }
}